Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -176,34 +206,80 @@ struct SymbolGraphLoader {
 
         return (symbolGraph, isMainSymbolGraph)
     }
+    
+    /// Adds the missing fallback and default availability information to the unified symbol graph
+    /// in case it didn't exists in the loaded symbol graphs.
+    private func addMissingAvailability(
+        unifiedGraph: inout UnifiedSymbolGraph,
+        unconditionallyUnavailablePlatformNames: [PlatformName],
+        registeredPlatforms: [PlatformName],
+        defaultAvailabilities: [DefaultAvailability.ModuleAvailability]
+    ) {
+        // The fallback platforms that are missing from the unified graph correspond to
+        // the fallback platforms that have not been registered yet,
+        // are not marked as unavailable,
+        // and the corresponding inheritance platform has a SGF (has been registered).
+        let missingFallbackPlatforms = DefaultAvailability.fallbackPlatforms.filter {
+            !registeredPlatforms.contains($0.key) &&
+            !unconditionallyUnavailablePlatformNames.contains($0.key) &&
+            registeredPlatforms.contains($0.value)
+        }
+        // Platforms that are defined in the Info.plist that had no corresponding SGF
+        // and are not being added as fallback of another platform.
+        let missingAvailabilities = defaultAvailabilities.filter {
+                !missingFallbackPlatforms.keys.contains($0.platformName) &&
+                !registeredPlatforms.contains($0.platformName)
+        }
+        let symbolsWithFallbackAvailability = unifiedGraph.symbols.compactMapValues { symbol  -> UnifiedSymbolGraph.Symbol in
```

Comment:
minor: Mapping all the values and then separately replacing the dictionary could result in potentially holding two versions of all the data for a brief moment unless the compiler can optimize it away. 

Since `UnifiedSymbolGraph.Symbol` is a class this mostly means holding two copies of a number of pointers but it could be avoided by directly iterating over the symbols and updating them in place.

Reply:
https://github.com/apple/swift-docc/commit/c55abc329895424275b96a3c47afef3f8c5bb3f9